.github/workflows/test.yml

```diff
@@ -70,6 +70,16 @@ jobs:
       #     flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
       #     # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
       #     flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+
+      #----------------------------------------------
+      #             check formatting
+      #----------------------------------------------
+      # Disabled until we agree to turn it on
+      # - name: Code formatting with black
+      #   run: |
+      #     # run black in check mode
+      #     # if files are not formatted correctly, the build will not go through
+      #     black . --check
```

Comment:
Can you confirm that flake8 does anything that `black --check` doesn't already do?

Reply:
From what I've gleaned, `black` is a code formatter and `flake8` is a PEP8 compliance checker. So running `black --check` would check if the code has been formatted per `black` configuration, and `flake8` would lint the code per the [PEP8 specification](https://pep8.readthedocs.io/en/latest/).

When I ran `flake8 schematic` on our core modules there were plenty of linting errors that came up, so it'll probably take a some time to fix the code per the error codes. Maybe it'd be a good idea to turn linting on after fixing the currently returned linting errors? I think we can turn on the black code formatter in the meantime, while we're fixing the linting errors.

Reply:
I just figured that once code is formatted by `black`, it's automatically PIP8-compliant. But I could be wrong, so having both can't hurt. You have a source describing this workflow, so I won't push on only using `black`. 

I'm guessing that most `flake8` linting errors will disappear once we format the code. A follow-up PR could address the remaining errors.

Reply:
Hm..as mentioned by many people in various [blog posts](https://luminousmen.com/post/my-unpopular-opinion-about-black-code-formatter), black does not fully remove all the PEP8 linting errors that come up when you run your code through a strict linter such as `flake8`. It removes a lot of the warnings and errors, but not all. So we have do one round of refactor even after running it through `black` before we enforce `flake8`. Not sure if I'm making sense, but we can discuss more about this during code review.

I'll mark this as resolved for now.